libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -399,44 +402,41 @@ TEST( TEST_IOT_UART, AFQP_IotUARTWriteAsyncReadAsyncLoopbackTest )
 TEST( TEST_IOT_UART, AFQP_AssistedIotUARTWriteAsync )
 {
     IotUARTHandle_t xUartHandle;
-    int32_t lIoctl, lWrite, lClose, lRead, lTransferAmount_1, lTransferAmount_2;
+    int32_t lIoctl, lWrite, lClose, lRead, lTransferAmount;
     uint8_t i;
     uint8_t cpBufferLarge[ testIotUART_BUFFER_LENGTH_LARGE + 1 ] = { 0 };
-    uint8_t cpBufferRead[ testIotUART_BUFFER_LENGTH_LARGE + 1 ] = { 0 };
-    uint8_t uStringCompare = 1;
+    BaseType_t xCallbackReturn;
 
     for (i = 0; i < testIotUART_BUFFER_LENGTH_LARGE; i++)
     {
         cpBufferLarge[i] = 0xAA;
     }
+    strncpy(&cpBufferLarge[testIotUART_BUFFER_LENGTH_LARGE ], "\n", 1);
```

Comment:
nit: you might just want:

```
cpBufferLarge[testIotUART_BUFFER_LENGTH_LARGE ] = '\n'
```

Reply:
okay

Reply:
No.  That is incorrect.  You want a NULL terminated string .